components/list/demo/list-drag-and-drop.js

```diff
@@ -0,0 +1,174 @@
+				primaryText: 'Glaciation',
+				supportingText: 'Supporting Info',
+				imgSrc: '',
+				items: []
+			}, {
+				key: '1-2',
+				primaryText: 'Weathering',
+				supportingText: 'Supporting Info',
+				imgSrc: '',
+				items: []
+			}, {
+				key: '1-3',
+				primaryText: 'Volcanism',
+				supportingText: 'Supporting Info',
+				imgSrc: '',
+				items: []
+			}]
+		}, {
+			key: '2',
+			primaryText: 'Flow and Transport Through Fractured Rocks',
+			supportingText: 'Fractures are ubiquitous in geologic media and important in disciplines such as physical and contaminant hydrogeology, geotechnical engineering, civil and environmental engineering, petroleum engineering among other areas.',
+			imgSrc: 'https://s.brightspace.com/course-images/images/e5fd575a-bc14-4a80-89e1-46f349a76178/tile-high-density-max-size.jpg',
+			items: [{
+				key: '2-1',
+				primaryText: 'Contaminant Transport',
+				supportingText: 'Supporting Info',
+				imgSrc: '',
+				items: []
+			}, {
+				key: '2-2',
+				primaryText: 'Modelling Flow in Fractured Media',
+				supportingText: 'Supporting Info',
+				imgSrc: '',
+				items: []
+			}]
+		}, {
+			key: '3',
+			primaryText: 'Applied Wetland Science',
+			supportingText: 'Advanced concepts on wetland ecosystems in the context of regional and global earth systems processes such as carbon and nitrogen cycling and climate change, applications of wetland paleoecology, use of isotopes and other geochemical tools in wetland science, and wetland engineering in landscape rehabilitation and ecotechnology.',
+			imgSrc: 'https://s.brightspace.com/course-images/images/38e839b1-37fa-470c-8830-b189ce4ae134/tile-high-density-max-size.jpg',
+			items: [{
+				key: '3-1',
+				primaryText: 'Carbon & Nitrogen Cycling',
+				supportingText: 'Supporting Info',
+				imgSrc: '',
+				items: []
+			}, {
+				key: '3-2',
+				primaryText: 'Wetland Engineering',
+				supportingText: 'Supporting Info',
+				imgSrc: '',
+				items: []
+			}]
+		}];
+	}
+
+	render() {
+		const renderList = (items, nested) => {
+			return html`
+				<d2l-list grid slot="${ifDefined(nested ? 'nested' : undefined)}">
+					${repeat(items, item => item.key, item => html`
+						<d2l-list-item
+							action-href="http://www.d2l.com"
+							draggable
+							drag-handle-text="${item.primaryText}"
+							drop-nested
+							key="${item.key}"
+							label="${item.primaryText}"
+							selectable>
+								${nested ? null : html`<img slot="illustration" src="${item.imgSrc}">`}
+								<d2l-list-item-content>
+									<div>${item.primaryText}</div>
+									<div slot="supporting-info">${item.supportingText}</div>
+								</d2l-list-item-content>
+								${item.items.length > 0 ? renderList(item.items, true) : null}
+						</d2l-list-item>
+					`)}
+				</d2l-list>
+			`;
+		};
+
+		return html`
+			<div @d2l-list-items-move="${this._handleListItemsMove}">
+				${renderList(this.items, false)}
+			</div>
+		`;
+	}
+
+	async _handleListItemsMove(e) {
```

Comment:
The consuming app is responsible for taking the item info (items being dragged and the item being dropped upon) and updating their data accordingly.  This can be somewhat complicated, but the logic very much depends on the consumer's data structure, and whether they are doing the move operation client or server-side.  In quizzing case, it is being done server-side.